### ISSUE TITLE
Turn off beeps

### ISSUE THREAD

**Issue by f2404** [linuxmint/Cinnamon#5253]
I have all effects disabled under Sound -> Sound Effects in the Settings, and the Effects Volume there is set to Muted. But I'm still hearing beeps in some applications (e.g. Nemo, Backgrounds settings) when pressing up (down) arrow button while the cursor is already at the topmost (bottommost) position.
How can I disable those beeps?

**Comment by mtwebster** [comment]
Check your accessibility settings, you might have some typing assistance turned on.

**Comment by f2404** [comment]
From all Accessibility settings, I only had Keyboard -> Enable audio alerts turned on. Nothing changes after turning it off, however.

**Comment by mtwebster** [comment]
What version of cinnamon are you running?

**Comment by f2404** [comment]
Cinnamon 3.0.1, Arch Linux.

**Comment by KickerTom** [comment]
Same problem here. Seems to really be connected to the new accessibility stuff, started after latest cinnamon upgrade. If I enable visual alerts, the screen flashes at the same time as the beep.
Happens also in e.g. firefox->find in page->enter text which is not found. Or Backspace in Nemo panel.

**Comment by mtwebster** [comment]
Can you try enabling and disabling the visual/audible alert stuff to see if it changes this?  I'll try to reproduce it later.

**Comment by KickerTom** [comment]
I tried, no change. Visual alert enable/disable works, audible not. Interesting is that the configuration of audible alert specifies bell.ogg file, and playing it in the configuration indeed plays a sound file, but the keyboard is always producing "speaker beep", not ogg file replay. Changing the sound file has no effect.

**Comment by KickerTom** [comment]
Hmm, after reboot, no alerts. So maybe just restart is needed after turning the audio alert off?

**Comment by leigh123linux** [comment]
@KickerTom  Try disabling the "nemo terminalprovider" plugin to fix the nemo sound issue

**Comment by KickerTom** [comment]
Thanks for getting back to me. Not sure what you mean by "disabling nemo terminalprovider plugin". If you mean a plugin in Nemo, I don't have such plugin installed. Just out of curiosity I tried disabling all plugins there and restarting Nemo, but it didn't make any difference.
I doubt this has anything to do with Nemo, as I already stated I can observe the same effect in e.g. Firefox, when you do Find in page (Ctrl+F) and search for text which does not exist.

**Comment by f2404** [comment]
I've accidentally found a work-around for this issue: unloading 'pcspkr' kernel module (`sudo rmmod pcspkr`) removes the beeps.

I'm not sure whether or not it is right to route speaker beeps through the sound card (as I was hearing them in my headphones), and which component (Cinnamon or not) is responsible for that...

**Comment by calestyo** [comment]
Is this the samen as what I've reported here? https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=824427

It should be noticed that this audible bell setting is apparently also inherited by programs running in cinnamon (e.g. firefox) which beeps everytime I search in a HTML page with no result.. pretty annoying...


@f2404: this is not really a solution, as it also disables the beeps for any legitimate uses e.g. when using the beep(1) command.

**Comment by mtwebster** [comment]
Cinnamon Settings-> Accessibility-> Keyboard-> Enable audio alerts should be controlling this.  Turn it off, these alerts should go away.

The triggering is handled internally by toolkits like Gtk - so any program using that toolkit will trigger the sound at certain events, by telling the x server to ring a bell, which our window manager sees, and does either the audio or visual alert or both, depending on what that setting is.

Turning off "Enable audio alerts" (and visual) tells our window manager to ignore these events.  I can't reproduce the issue here on my end in Mint - I will give it a go on my arch vbox later.

Note: Any program can 'subscribe' to these triggers system-wide, and circumvent our controls.

**Comment by calestyo** [comment]
As I've written in the Debian bug, I have them all set to off, yet it still beeps. :(

**Comment by mtwebster** [comment]
Would you guys mind running this script: https://gist.github.com/mtwebster/d9559821b6fb0f948f88f65e0ea9146d

Download it, or paste it into a text file, make it executable (`chmod a+x *filename*`)...

This might help me find out what's causing this problem - while it's running, do whatever you'd normally do to cause the sounds to occur, then pastebin the output.

Thanks

**Comment by calestyo** [comment]
```
$ ./sound-source-mon 
Monitoring sound stream...  ctrl-c to terminate


Stream added:   1 None alsa_output.pci-0000_00_1b.0.analog-stereo Built-in Audio Analog Stereo /devices/pci0000:00/0000:00:1b.0/sound/card0
Stream added:   2 None alsa_input.pci-0000_00_1b.0.analog-stereo Built-in Audio Analog Stereo None
Stream added:   3 None mpv Media Player 194.mp4 - mpv None
Stream added:   4 None System Sounds None None
```

Doesn't print more than that, even after restarting firefox.
And I did trigger the beeps from within firefox... (and also from gnome-terminal, though I had to enable the terminal bell setting for that, which I disabled since this issue rose up in cinnamon).

What I further just noticed:
When I Ctrl-F in Firefox and search a string that doesn't exist to trigger the beep... it doesn't happen always, that is:
When I search on a site for e.g. "fo" and that string still exists and then I add another "f" to make it "fof" it won't beep the first time.
When I add further characters like "fofgggg" it generally won't beep again until I go back to a string that it can find (i.e. "fo").
So when I do that (going back to "fo") and then wait sufficiently long (2-3s) I won't beep even if I add the "f" again (and it doesn't find "fof"... but when I do it pretty fast like
fo -> f, backspace, f, backspace, f...
it will beep every time.

Not sure if that helps in any way.

**Comment by mtwebster** [comment]
Does anything get output if you do something like turn the volume up or down or change workspaces with a hotkey (try enabling a few things in the sound effects page of sound settings)

Try enabling audio alerts from accessibility settings - do you then get 2 sounds each time you trigger it?  Or still just one?  Is it the bell.ogg sound or something else?  With audio alerts enabled, you should see this spewed out in that little monitor program i made.

I just want to be sure the utility is working for you the same way it's working for me.

**Comment by mtwebster** [comment]
by the way, it looks like firefox can also be individually disabled in about:config - *accessibility.typeaheadfind.enablesound*

**Comment by calestyo** [comment]
```
I did for minimizing windows and switching workspaces:
$ ./sound-source-mon 
Monitoring sound stream...  ctrl-c to terminate


Stream added:   1 None alsa_output.pci-0000_00_1b.0.analog-stereo Built-in Audio Analog Stereo /devices/pci0000:00/0000:00:1b.0/sound/card0
Stream added:   2 None alsa_input.pci-0000_00_1b.0.analog-stereo Built-in Audio Analog Stereo None
Stream added:   3 None mpv Media Player 194.mp4 - mpv None
Stream added:   4 org.gnome.VolumeControl cinnamon Peak detect None
Stream added:   5 None System Sounds None None
Stream added:   6 org.freedesktop.libcanberra libcanberra /usr/share/sounds/freedesktop/stereo/bell.oga None
Stream removed: 6
Stream added:   7 org.freedesktop.libcanberra libcanberra /usr/share/sounds/freedesktop/stereo/bell.oga None
Stream removed: 7
Stream added:   8 org.freedesktop.libcanberra libcanberra /usr/share/sounds/freedesktop/stereo/bell.oga None
Stream removed: 8
```

(btw: Switch workspace sounds only work when doing it via workspace switcher applet, not when doing it via hotkeys)...

Anyway one sees some events in your program.

But just to emphasise this again:
The bell sound I hear in Firefox/etc. is really the pcspkr beeper, I guess... and not somthing from the soundcard as in your the sound events for minimising windows.

To mute those sounds right now (since this whole issue appeared) I mute the respective channel  ("Beep") in alsamixer for the soundcard device.

**Comment by mtwebster** [comment]
Man.. an actual speaker beep.  I am mixing your reporting with the other user who said it was in the headphones.. I'm not sure what could be causing this.  Have you had a look at https://wiki.archlinux.org/index.php/Disable_PC_speaker_beep?  Check your running processes for clues?  Any type of alert like this coming from Cinnamon, you'd see it in that monitor utility, so I think something else must be causing it.

**Comment by mtwebster** [comment]
I'm particularly curious if adding:
```
[Settings]
gtk-error-bell = 0
```
to ~/.config/gtk-3.0/settings.ini has any effect here.

We don't appear to control that with our usual machinery (in fact gtk doesn't provide a way to control it system-wide, either, like many other similar settings).

**Comment by mtwebster** [comment]
Actually, if this does solve it, I have an api I wrote that I can automatically update that file for you via cinnamon settings (we had to do this to enable dark theme variants, which suffers from the same problem of not being configurable system-wide without having root permissions)

**Comment by calestyo** [comment]
Sorry for the confusion I might have caused... shall we open a new bug for this then?
Anyway as for what you've wrote above:
The arch wiki also suggests the setting which I already have set to false (and doesn't affect this apparently)... unloading the speaker module is IMHO not an alternative but just a bad workaround like muting the channel.

But... you had a good guess with gtk-error-bell.. I set it retried with firefox,... didn't change anything there (not sure why... perhaps firefox is doing this completely independent?).
Then I re-checked with gedit, i.e. also doing a search there, but gedit - on must notice - apparently never rings the bell when no match is found (like firefox), but it *does* when you have the search field open and backspace at the empty field (i.e. it tells you "I'm already empty")...
Now when I set gtk-error-bell, it in fact no longer beeps then in gedit.

Not sure if this was a change then in Debian's gtk, at least it happened to start when I upgraded cinnmon.

Also not sure what about firefox... of course I can use the setting you found out above, but if we knew whether it tries to inherit that somehow from the desktop environment, it would perhaps mean another fix is needed for FF.

**Comment by mtwebster** [comment]
No, this bug is fine to keep open for now.. Firefox could very well just ignore the gtk setting - I know they recently updated to using gtk3 (I think?) but they're under no obligation to abide by GtkSettings, particularly if they already had their own machinery/systems in place - what still puzzles me, however, is even in my version of gtk3 here (on Mint 18), the gtk-error-bell setting also defaults to True, yet I don't suffer from this issue.  In fact, if I turn it off, none of my nice shiny new audible/visual alert settings work at all.

If you care to, could you maybe make a new user and see if the issue occurs in that session also?  I'm trying to understand why this appears to be so under-reported, for how annoying it might be.  Maybe simply that support for these internal speakers is spotty?  Or newer PCs don't have one anymore?  I'm not sure - some vestigial feature buried in X server?

**Comment by f2404** [comment]
> Man.. an actual speaker beep. I am mixing your reporting with the other user who said it was in the
> headphones.. 
I guess I was hearing that in the headphones because my laptop simply does not have a pc speaker, so its sounds get routed to the sound card. Just a guess, but proved with the fact that unloading pcspkr kernel module removes the sounds.

**Comment by calestyo** [comment]
@mtwebster I just tried with a new user, and the issues occurs as well.

Perhaps some speakers are either not exported through the soundcard via ALSA, or the default to be muted?

**Comment by calestyo** [comment]
fyi: still in Cinnamon 3.2.6

**Comment by dg1727** [comment]
I'm having a similar experience as user "calestyo".  All the beeps in question are PC-speaker beeps, **not** line-out audio-file playback:  
- Bluefish was beeping when I backspaced in an empty search field, but the ~/.config/gtk-3.0/settings.ini stopped that.  I wish there were also an option to select an audio file to be played for this beep; I looked briefly in the GTK+ documentation on the Web, but I didn't find such an option.  
- Firefox beeps when I type a search string that doesn't exist in the webpage.  I've set _accessibility.typeaheadfind.enablesound_ to false, and I assume the beeps will go away after I restart Firefox.  I also noticed _accessibility.typeaheadfind.soundURL_ which should let Firefox use the sound card as opposed to the PC speaker.  
- There is a beep on logout of the desktop (e.g., restarting the PC).  It seems louder and/or longer than the error beeps in text fields.  I'd rather be able to disable this beep specifically, and optionally select an audio file to be played on the sound card's line out, instead of disabling all PC-speaker beeps altogether.  

LMDE 2 Cinnamon 64-bit, Cinnamon 3.2.7

**Comment by dg1727** [comment]
Yes, after restart of Firefox, it's no longer beeping when I type a search string that doesn't exist in the webpage.

**Comment by dg1727** [comment]
To select an audio file for the X11 beep, a technique in the following bug report worked for me:  
https://bugs.launchpad.net/ubuntu/+source/pulseaudio/+bug/231234
Uncomment the lines in /etc/pulse/default.pa that are as follows:  (they're typically not consecutive) 
`load-sample-lazy x11-bell FILENAME`
`load-module module-x11-bell sample=x11-bell`

This affects programs that can produce an X11 beep, including GTK+ programs such as Bluefish, but it doesn't affect the beep on logout.

**Comment by selliott512** [comment]
It's not an ideal solution, but one way to get rid of such beeps is to find it's audio file and then replace that audio file with silence.  For example, on my Fedora 25 system the beep that's played, both in `gedit` when the end of the buffer is reached and in `Eclipse` in some cases, is `drip.ogg`. On my system it's in `/usr/share/sounds/gnome/default/alerts`. Maybe it works to simply remove it, but I chose to replace it with a silent file of the same type and length:
```shell
cd /usr/share/sounds/gnome/default/alerts
mv drip.ogg drip.ogg.orig
sox -v 0 drip.ogg.orig silence.ogg
ln -s silence.ogg drip.ogg
```

After the above on my system `/usr/share/sounds/gnome/default/alerts` looks like:
```
-rw-r--r--. 1 root root 13322 Oct 21 12:22 bark.ogg
lrwxrwxrwx. 1 root root    11 Mar  8 11:03 drip.ogg -> silence.ogg
-rw-r--r--. 1 root root  8495 Oct 21 12:22 drip.ogg.orig
-rw-r--r--. 1 root root 18999 Oct 21 12:22 glass.ogg
-rw-r--r--. 1 root root  4065 Mar  8 11:03 silence.ogg
-rw-r--r--. 1 root root 20011 Oct 21 12:22 sonar.ogg
```

Alternatively the beep could be made quieter with something other than `-v 0`.  Maybe `-v 0.3` or something like that.

**Comment by calestyo** [comment]
o.O ... or I dismount my soundcard?!

**Comment by HW71** [comment]
Hi all,

actually for me the hint @mtwebster gave here:

https://github.com/linuxmint/Cinnamon/issues/5253#issuecomment-242879319

did do the trick: in my case this ugly "beep" coming from the internal PC speaker occurred for example when pressing [Backspace] more often than characters were available in any textfield (for example password field in Legitimation dialog after starting Synaptic).

Now creating a new `settings.ini` and adding `gtk-error-bell = 0` as outlined above finally eliminated this "beep" without removing `pcspkr` kernel module (I didn't want to do that, because I'm afraid that this could have side effects, for example not alerts in case of important hardware issues).

After restarting my laptop the setting became effective (but I assume logging-off and on again would have been sufficient).

HTH & Cheers, Erik

**Comment by RoyiAvital** [comment]
Does it affect MATLAB's sound (Usually a beep) when there is a run error? 

In Windows MATLAB makes a beep for error.
Yet on Linux Mint I can't hear it.

MATLAB's `beep()` status is ON yet no sound.

Thank You.

**Comment by SEV007** [comment]
Actually it's pretty simple.
Just go to Keyboard >> Shortcuts >> Sound and Media (on the left panel) >> Rewind
Now all you have to do is unassign the sound effect by double clicking it and pressing ESCAPE ;)
And we are done. No more backspace beeping sound. Enjoy.

**Comment by linkp** [comment]
@SEV007 What you describe is the steps to unassign physical keys that control media player software.

**Comment by DanFraserUK** [comment]
Open dconf editor, search for nemo or go to /org/nemo/extensions/nemo-terminal and turn off "audible-bell".

**Comment by digiberk** [comment]
It seems this problem is still present, I just did a fresh install of Fedora 31 with Cinnamon 4.4.6 and the issue is still there. When I hit backspace in the search field for "System Settings", it would beep. It gave me a shock in the beginning, haha! :sob:

**Comment by basmevissen** [comment]
This seems to happen when you switch off _all_ sound effects. I just blacklisted the PC speaker to work around it. Bug is definitly still there (running F31 as well).

**Comment by lmarszalek** [comment]
Worked for me:
Go to Settings/Sound/Sound effects tab
"Notifications" is the only one unassigned by default.
Set it to /usr/share/cinnamon-control-center/sounds/notification.oga as it should be.
Done. Fedora 31 Cinnamon 4.4.8

**Comment by benradey** [comment]
I was having trouble with the actual PC speaker beeping (not played through the sound card) when I would try to backspace in an empty terminal or a firefox search with no results.

Here's what seems to have solved it for me in dconf-editor:

org.cinnamon.settings-daemon.peripherals.keyboard bell-mode 'off'

Fedora 31 with Cinnamon 4.4.8

**Comment by calestyo** [comment]
Just tried it an can confirm, setting it to `off` while at the same time enabling the channel for "Beep" in my ALSA mixer again let's me hear the pcspkr beep when I run e.g. the beep binary, but doesn't give me any annoying beep when e.g. searching in firefox.

I've then noticed the following:
In the cinnamon control centre, System Settings, Keyboard.
When I change the switch for "Enable audio alerts" dconf-editor asks whether I want to reload `org.cinnamon.settings-daemon.peripherals.keyboard.bell-mode` cause it would have been changed.
But when I do and regardless of the switch position in the control centre, it's value is always `on`.

So maybe the bug is as simple as this, both set it accidentally to `on`?

**Comment by melsophos** [comment]
> Here's what seems to have solved it for me in dconf-editor:
> org.cinnamon.settings-daemon.peripherals.keyboard bell-mode 'off'

This worked for me, thanks!

> e.g. searching in firefox.

Concerning the beep from the searchbar in Firefox, I thought this does not have the same origin since it can be disabled by setting `accessibility.typeaheadfind.enablesound` to false.

**Comment by xss** [comment]
In case you are on Ubuntu with Cinnamon, and initially simply installed Ubuntu with its default Desktop Environment, the annoying beep sound might be some left-over from Ubuntu's own desktop environment's sound theme. There doesn't seem a way to disable this from within Cinnamon or dconf-editor. Also changing the above mentioned `bell-mode` settings to `'off'` didn't do anything for me. I identified the following audio file as the annoying bell I kept hearing:

    /usr/share/sounds/ubuntu/stereo/bell.ogg

The following worked for me:
Log out of your Cinnamon session. From the Login Screen, select `Ubuntu (Default)` as desktop environment, then log into Ubuntu with your normal user. Open Ubuntu's `Settings` applet (press `SUPER`/`WINDOWS` key and type "`settings`"), browse to `Sound` (left-hand navigation), select tab `Sound Effects`, and there disable the switch on the right of `Alert volume`:

[Screenshot](https://imgur.com/f3hxXJt)

Close the window and log out of this Ubuntu session. Back on Login Screen, select Cinnamon and log back in. This should do it, no more bell sounds in some random applications.

I assume there is a corresponding dconf-editor setting for the above switch that would make it unnecessary to log in with another Desktop Environment. However, I was unable to identify the setting. Feel free to let me know. :)
